Source/ACE/Entity/Creature.cs

```diff
@@ -587,5 +586,66 @@ public void DoWorldBroadcast(string message, ChatMessageType messageType)
 
             WorldManager.BroadcastToAll(sysMessage);
         }
+
+        /// <summary>
+        /// This signature services MoveToObject and TurnToObject
+        /// Update Position prior to start, start them moving or turning, set statemachine to moving.
+        /// Moved from player - we need to be able to move creatures as well.   Og II
+        /// </summary>
+        /// <param name="worldObjectPosition">Position in the world</param>
+        /// <param name="sequence">Sequence for the object getting the message.</param>
+        /// <param name="movementType">What type of movement are we about to execute</param>
+        /// <param name="targetGuid">Who are we moving or turning toward</param>
+        /// <returns>MovementStates</returns>
+        public void OnAutonomousMove(Position worldObjectPosition, SequenceManager sequence, MovementTypes movementType, ObjectGuid targetGuid)
+        {
+            UniversalMotion newMotion = new UniversalMotion(MotionStance.Standing, worldObjectPosition, targetGuid);
+            newMotion.DistanceFrom = 0.60f;
+            newMotion.MovementTypes = movementType;
+            CurrentLandblock.EnqueueBroadcast(Location, Landblock.MaxObjectRange, new GameMessageUpdatePosition(this));
+            CurrentLandblock.EnqueueBroadcastMotion(this, newMotion);
+        }
+
+        /// <summary>
+        /// This is the OnUse method.   This is just an initial implemention.   I have put in the turn to action at this point.
+        /// If we are out of use radius, move to the object.   Once in range, let's turn the creature toward us and get started.
+        /// Note - we may need to make an NPC class vs monster as using a monster does not make them turn towrad you as I recall. Og II
+        ///  Also, once we are reading in the emotes table by weenie - this will automatically customize the behavior for creatures.
+        /// </summary>
+        /// <param name="playerId">Identity of the player we are interacting with</param>
+        public override void HandleActionOnUse(ObjectGuid playerId)
+        {
+            ActionChain chain = new ActionChain();
+            CurrentLandblock.ChainOnObject(chain, playerId, wo =>
+            {
+                Player player = wo as Player;
+                if (player == null)
+                {
+                    return;
+                }
+
+                if (!player.IsWithinUseRadiusOf(this))
+                    player.DoMoveTo(this);
```

Comment:
Does this actually work?  If you "Use" something while your away, you should move to it, then "Use" it.  From this code it looks like your just doing the motion and not the use.

Reply:
Yes it works.   I just put in the move to and turn.   I need to read the expected behavior from the weenie that does not exist yet.   Just reall prof of concept but it does work and is a building block for what I need to do next

Reply:
yea after you reach your destination it calls back and then does the rest of the actions.   Sorry misunderstood the question.

Reply:
It works, I use t in vendor and my guy will walk to the vendor and then then the I use fires and it opens